common/lib/xmodule/xmodule/tests/test_license.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+"""Test for License classes."""
+import unittest
+from xmodule.license import parse_license, License, ARRLicense, CCLicense
+
+
+class BaseLicenseTest(unittest.TestCase):
+    """Base test class for License tests"""
+    def assert_equal_license(self, lic_1, lic_2):
+        """
+        Asserts that lic_1 and lic_2 are both licenses of the same
+        type, with the same versions as applicable.
+        """
+        # Assert that lic_1 and lic_2 are both License types,
+        # otherwise further checks will fail
+        self.assertTrue(isinstance(lic_1, License))
+        self.assertTrue(isinstance(lic_2, License))
+        # Assert licenses are of the same type
+        self.assertEqual(lic_1.kind, lic_2.kind)
+        # If they're not of the ARR type license, they're versioned, so
+        # assert that their versions are also equal
+        if not lic_1.kind == "ARR":
+            self.assertEqual(lic_1.version, lic_2.version)
+
+class LicenseTest(BaseLicenseTest):
+    """Tests for License class."""
+
+    def setUp(self):
+        self.no_license = License()
+        self.default_license = ARRLicense()
+        self.random_license = License('RAND', "v2.56")
+        self.random_dict = {"kind": "RAND", "version": "v2.56"}
+        self.invalid_dict = {"kind": "RAND"}
+
+    def test_html(self):
+        """Should never be called in production, but otherwise say it's not licensed."""
+        self.assertEqual(self.no_license.html, u"<p>This resource is not licensed.</p>")
```

Comment:
A: Default license renders ARRLicense, this would never get rendered. With the templates this is clearer in the code as well.